cs_tools/cli/tools/_scriptability/app.py

```diff
@@ -0,0 +1,363 @@
+
+            objects = r.json().get('object', [])
+            for _ in objects:
+                status = _['info']['status']
+                if not status['status_code'] == 'OK':  # usually access errors.
+                    console.log(f"[bold red]unable to get {_['info']['name']}: {_['info']['status']}[/]")
+
+                else:
+                    console.log(f"{_['info']['filename']} (OK)")
+                    _write_tml_obj_to_file(path=path, tml=_)
+
+    else:  # getting associated, so get the full pack.
+        with console.status(f"[bold green]exporting ${export_ids} with associated content.[/]"):
+            object_list = ts.metadata.get_edoc_object_list(export_ids)
+            r = ts.api._metadata.edoc_export_epack(
+                {
+                    "object": object_list,
+                    "export_dependencies": True
+                }
+            )
+            console.log(r)
+            _write_tml_package_to_file(path=path, contents=r.json()['zip_file'])
+
+
+def _write_tml_obj_to_file(path: pathlib.Path, tml: str) -> None:
+    """
+    Writes the TML to a file.
+    :param path: The path to write to.  Can be a directory or filename.  If it's a directory, the file will be saved
+    in the form <GUI>.<type>.TML
+    :param TML:  The TML as a JSON object returned from ThoughtSpot
+    :return: None
+    """
+    guid = tml['info']['id']
+    type = tml['info']['type']
+    name = tml['info']['name']
+
+    fn = f"{guid}.{type}.tml"
+    if path:
+        fn = f"{path}/{fn}"
+
+    console.log(f'writing {name} to {fn}')
+    with open (fn, "w") as f:
+        f.write(tml['edoc'])
+
+
+def _write_tml_package_to_file(path: pathlib.Path, contents: str) -> None:
+    """
+    :param path: The path to write to.  Can be a directory or filename.  If it's a directory, the file will be saved
+    in the form <GUI>.<type>.TML
+    :param contents: The contents to write to the zip file.
+    :return: None
+    """
+    now = datetime.datetime.now().strftime("_%Y%m%d_%H%M%S")
+    filepath = path / f"{now}.scriptability.zip"
+    base64_to_file(contents, filepath=filepath)
+
+    try:
+        with ZipFile(filepath, 'r') as zippy:
+            for info in zippy.infolist():
+                with zippy.open(info.filename) as tml_file:
+                    if (info.filename.endswith('tml')):
+                        (filename, tml_type, _) = info.filename.split('.', maxsplit=3)
+                        console.log(f'writing {filename}')
+                        tml_obj = YAMLTML.get_tml_object(tml_file.read().decode('utf-8'))
+                        tml_outfile = path / f"{tml_obj.guid}.{tml_obj.content_type}.tml"
+                        with open(tml_outfile, 'w') as outfile:
+                            outfile.write(YAMLTML.dump_tml_object(tml_obj))
+    except Exception as err:
+        console.stderr(f"[bold red]{err}[/]")
+
+
+@app.command(cls=CSToolsCommand)
```

Comment:
`@app.command(name='import', cls=CSToolsCommand)`

if you wanted to call this `import` for the users. then your function can be called `def import_(...)`

up to you!

Reply:
This is great.  Switched to import to match the API and docs.